app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+
+  validates :name, presence: true
+  validates :photo, presence: true
+  validates :bio, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  def three_recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+end
```

Comment:
- [ ] After setting up `devise` properly, please remember to activate the necessary modules based on the requirements for this milestone in the `app/models/user.rb`. This will ensure that all required functionalities for user authentication are setup properly. 💻 😎